api/saml.go

```diff
@@ -0,0 +1,50 @@
+package api
+
+import (
+	"encoding/xml"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/crewjam/saml"
+	"github.com/crewjam/saml/samlsp"
+)
+
+// getSAMLServiceProvider generates a new service provider object with the
+// (optionally) provided descriptor (metadata) for the identity provider.
+func (a *API) getSAMLServiceProvider(identityProvider *saml.EntityDescriptor) *saml.ServiceProvider {
+	externalURL, err := url.ParseRequestURI(a.config.API.ExternalURL)
+	if err != nil {
+		panic(err)
```

Comment:
should we return an error and handle it instead?

Reply:
Technically this would not fail as `Validate` in the config will take care of it.

Reply:
Changed the validation logic as it really wasn't there, added a comment here.